Packs/TrendMicroVisionOne/Integrations/TrendMicroVisionOneEventCollector/TrendMicroVisionOneEventCollector.py

```diff
@@ -511,89 +424,83 @@ def get_datetime_range(
-
-        latest_log_time = latest_log_time_datetime.strftime(date_format)
+        latest_log_time = max(logs, key=lambda log: datetime.strptime(log[created_time_field], date_format))[created_time_field]
         demisto.info(f'{latest_log_time=} for {log_type=}')
-        return latest_log_time_datetime.strftime(date_format)
+        return latest_log_time
 
-    demisto.info(f'No new logs for {log_type=}')
+    demisto.info(f'No logs found for {log_type=}')
     return ''
 
 
-def get_all_latest_time_logs_ids(
-    logs: List[Dict],
+def get_all_latest_logs_ids(
+    logs: List[dict],
     log_type: str,
     log_id_field_name: str,
     log_created_time_field_name: str = '_time',
-    date_format: str = DATE_FORMAT
-) -> Tuple[List[str], str]:
+    date_format: str = DATE_FORMAT,
+    latest_log_time: str = None
+) -> tuple[List[str], str]:
     """
-    Get all the logs that their time is equal to the last log that occurred.
+    Get all the logs that their time is equal to the last or earliest log that occurred.
 
     Args:
         logs (list): a list of logs.
         log_type (str): the log type
         log_id_field_name (str): the id field name of the log type
         log_created_time_field_name (str): the created time field of the log
         date_format (str): the date format of the logs
+        latest_log_time (str): the latest log time from the logs list
 
     Returns: all the logs their created time is equal to the latest created time log & latest log time
 
     """
-    latest_occurred_log = get_latest_log_created_time(
+    latest_occurred_time_log = latest_log_time or get_latest_log_created_time(
         logs=logs,
         log_type=log_type,
         created_time_field=log_created_time_field_name,
-        date_format=date_format,
+        date_format=date_format
     )
 
     # if there aren't any new logs, no need to cache anything
-    if not latest_occurred_log:
-        return [], latest_occurred_log
+    if not latest_occurred_time_log:
+        return [], latest_occurred_time_log
 
     latest_occurred_time_log_ids: Set[str] = set()
 
     for log in logs:
-        if log.get(log_created_time_field_name) == latest_occurred_log:
-            if log_id := log.get(log_id_field_name):
-                demisto.info(f'adding log with ID {log_id} to latest occurred time logs')
-                latest_occurred_time_log_ids.add(log_id)
+        if log.get(log_created_time_field_name) == latest_occurred_time_log and (log_id := log.get(log_id_field_name)):
+            demisto.info(f'adding log with ID {log_id} to latest occurred time logs')
+            latest_occurred_time_log_ids.add(log_id)
 
-    demisto.info(f'{latest_occurred_time_log_ids=}')
-    return list(latest_occurred_time_log_ids), latest_occurred_log
+    demisto.info(f'{latest_occurred_time_log_ids=} for {log_type=}')
+    return list(latest_occurred_time_log_ids), latest_occurred_time_log
 
 
 def dedup_fetched_logs(
```

Comment:
lets make the logs in this function debug logs

Reply:
to do anything in debug log in xsiam will not be seen in the logs when debugging customer's environment so i really prefer to keep it info.

why its like this? that's a good question